src/components/views/messages/MVideoBody.tsx

```diff
@@ -259,13 +267,20 @@ export default class MVideoBody extends React.PureComponent<IBodyProps, IState>
                     ref={this.videoRef}
                     src={contentUrl}
                     title={content.body}
-                    controls
+                    controls={controls}
+                    loop={loop}
                     preload={preload}
                     muted={autoplay}
                     autoPlay={autoplay}
                     height={height}
                     width={width}
                     poster={poster}
+                    onMouseOver={!autoplay && !controls ? event => (event.target as HTMLVideoElement).play() : () => { }}
+                    onMouseOut={!autoplay && !controls ?
+                        event => {
+                            (event.target as HTMLVideoElement).pause();
+                            (event.target as HTMLVideoElement).currentTime = 0;
+                        } : () => { }}
```

Comment:
I think this would be easier to read if these were separate methods somewhere above